public/app/features/dashboard-scene/sharing/ExportButton/ExportAsImage.tsx

```diff
@@ -0,0 +1,179 @@
+import { css } from '@emotion/css';
+import { saveAs } from 'file-saver';
+import { useEffect } from 'react';
+import { useAsyncFn } from 'react-use';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Trans, t } from '@grafana/i18n';
+import { config } from '@grafana/runtime';
+import { SceneComponentProps, SceneObjectBase } from '@grafana/scenes';
+import { Alert, Button, useStyles2 } from '@grafana/ui';
+import { DashboardInteractions } from 'app/features/dashboard-scene/utils/interactions';
+import { getDashboardSceneFor } from 'app/features/dashboard-scene/utils/utils';
+
+import { ImagePreview } from '../components/ImagePreview';
+import { SceneShareTabState, ShareView } from '../types';
+
+import { generateDashboardImage } from './utils';
+
+export class ExportAsImage extends SceneObjectBase<SceneShareTabState> implements ShareView {
+  static Component = ExportAsImageRenderer;
+
+  public getTabLabel() {
+    return t('share-modal.image.title', 'Export as image');
+  }
+}
+
+function ExportAsImageRenderer({ model }: SceneComponentProps<ExportAsImage>) {
+  const { onDismiss } = model.useState();
+  const dashboard = getDashboardSceneFor(model);
+  const styles = useStyles2(getStyles);
+
+  const [{ loading: isLoading, value: imageBlob, error }, onExport] = useAsyncFn(async () => {
+    try {
+      const result = await generateDashboardImage({
+        dashboard,
+        scale: config.rendererDefaultImageScale || 1,
+      });
+
+      if (result.error) {
+        throw new Error(result.error);
+      }
+
+      DashboardInteractions.generateDashboardImageClicked({
+        scale: config.rendererDefaultImageScale || 1,
+        shareResource: 'dashboard',
+        success: true,
+      });
+
+      return result.blob;
+    } catch (error) {
+      console.error('Error exporting image:', error);
+      DashboardInteractions.generateDashboardImageClicked({
```

Comment:
Do we need to track image errors with rudderstack? Can we  Faro, instead?

Reply:
using rudderstack as that is how we currently instrument analytics for features in squad :)